src/ape/contracts/base.py

```diff
@@ -881,6 +881,30 @@ def get_event_by_signature(self, signature: str) -> ContractEvent:
 
         raise err
 
+    def get_error_by_signature(self, signature: str) -> CustomErrorType:
+        """
+        Get an error by its signature, similar to
+        :meth:`~ape.contracts.ContractInstance.get_event_by_signature`.
+
+        Args:
+            signature (str): The signature of the error.
+
+        Returns:
+            :class:`~ape.types.CustomErrorType`
+        """
+
+        name_from_sig = signature.split("(")[0].strip()
+        options = self._errors_.get(name_from_sig, [])
+        err = ContractError(f"No error found with signature '{signature}'.")
```

Comment:
Wonder if this exception is too ambigiously named :thinking:

At least something to consider for breaking changes in 0.7.0

Reply:
Add thoughts to APE-822, just made the ticket